pkg/api/controller.go

```diff
@@ -802,27 +810,71 @@ func (c *Controller) ListUsers(w http.ResponseWriter, r *http.Request, params Li
 	writeResponse(w, http.StatusOK, response)
 }
 
+func (c *Controller) generateResetPasswordToken(email string) (string, error) {
+	secret := c.Auth.SecretStore().SharedSecret()
+	currentTime := time.Now()
+	return GenerateJWTResetPassword(secret, email, currentTime, currentTime.Add(DefaultResetPasswordExpiration))
+}
+
+func (c *Controller) inviteUserRequest(email string) error {
+	// TODO (@shimi9276) create template for sending the email with link for invite using the passed template
+	token, err := c.generateResetPasswordToken(email)
+	if err != nil {
+		return err
+	}
+	err = c.Emailer.SendEmailWithLimit([]string{email}, token, token, nil)
+	if err != nil {
+		return err
+	}
+	c.Logger.WithField("email", email).Info("invite email sent")
+	return nil
+}
+
 func (c *Controller) CreateUser(w http.ResponseWriter, r *http.Request, body CreateUserJSONRequestBody) {
+	invite := swag.BoolValue(body.InviteUser)
+	id := body.Id
+	var email *string
+	if invite {
+		addr, err := mail.ParseAddress(body.Id)
+		if err != nil {
+			c.Logger.WithError(err).WithField("userID", id).Warn("failed parsing email")
+			writeError(w, http.StatusBadRequest, "Invalid email format")
+			return
+		}
+		id = strings.ToLower(addr.Address)
+		email = &body.Id
+	}
 	if !c.authorize(w, r, permissions.Node{
 		Permission: permissions.Permission{
 			Action:   permissions.CreateUserAction,
-			Resource: permissions.UserArn(body.Id),
+			Resource: permissions.UserArn(id),
 		},
 	}) {
 		return
 	}
 	u := &model.User{
 		CreatedAt:    time.Now().UTC(),
-		Username:     body.Id,
+		Username:     id,
 		FriendlyName: nil,
 		Source:       "internal",
+		Email:        email,
 	}
 	ctx := r.Context()
 	c.LogAction(ctx, "create_user")
 	_, err := c.Auth.CreateUser(ctx, u)
+
 	if handleAPIError(w, err) {
+		c.Logger.WithError(err).WithField("email", id).Warn("failed creating user")
```

Comment:
This is the general flow - "user_id"

Reply:
At this point id == u.ID, but I do agree that for readability sake we should pass u.ID. Fixed